test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp

```diff
@@ -2152,3 +2156,151 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator, FromEn
         EXPECT_TRUE(static_cast<const PositionDerivative&>(*propagator.getDynamics()[4]).isDefined());
     }
 }
+
+class OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator_Data_Success
+    : public OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator,
+      public ::testing::WithParamInterface<Tuple<Instant, Instant, Duration, String>>
+{
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    Input_Data_Spans_Valid,
+    OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator_Data_Success,
+    testing::Values(
+        std::make_tuple(
+            Instant::Now() - Duration::Weeks(52 * 5),
+            Instant::Now() - Duration::Weeks(52 * 4.9),
+            Duration::Days(7),
+            "NRLMSISE Past Data limit pass"
+        ),
+        std::make_tuple(
+            Instant::Now() + Duration::Weeks(52 * 0.9),
+            Instant::Now() + Duration::Weeks(52 * 1.0),
+            Duration::Days(7),
+            "IERS Finals2000A Future Data limit pass"
+        )
+    )
+);
+
+TEST_P(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator_Data_Success, EdgeCases_Input_Data_Horizon)
+{
+    // long-horizon, low granularity propagation to test input data availability
+
+    // currently limited by:
+    // -5 years in the past due to CSSI space weather data when computing drag
+    // +1 year in the future due to finals2000a prediction span when computing polar motion for frame conversions
+
+    auto parameters = GetParam();
+
+    Instant startInstant = std::get<0>(parameters);
+    Instant endInstant = std::get<1>(parameters);
+    Duration step = std::get<2>(parameters);
```

Comment:
```suggestion
    const Instant startInstant = std::get<0>(parameters);
    const Instant endInstant = std::get<1>(parameters);
    const Duration step = std::get<2>(parameters);
```
God why couldn't const be the default. #Rust